### PR TITLE
Initial Support for SunPKCS11 provider in native image

### PR DESCRIPTION
**Pull request by sberyozkin**
At the moment this PR only registers `SunPKCS11` as an additional security provider (which ensures it is available in native image) and starts with a simple integration test. I'm not sure yet if it sufficient to address #21099 as it is hard to automate such tests. In general it is worth starting investing more time into `SunPKCS11`, as it can be used for default FIPS configuration as well.

### CODE REVIEW

**Review by gsmet — CHANGES_REQUESTED** [review]

File: integration-tests/sunpkcs11/pom.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-integration-tests-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-integration-test-sunpkcs11</artifactId>
+    <name>Quarkus - Integration Tests - Sun PKCS 11</name>
```
[the remark this answers is not in the document]
Comment (sberyozkin) [reply integration-tests/sunpkcs11/pom.xml:13]: I was hoping to expand but it does not seem possible to add various cryptoki PKCS11 implementations, I can probably update a bouncycastle test to register both BC and SunPKCS11 for a simple test like this one.

[the remark this answers is not in the document]
Comment (sberyozkin) [reply integration-tests/sunpkcs11/pom.xml]: I updated the BC test, it is actually better this way as it also tests more than one configured provider value, in fact this specific combination of providers is used in this [tutorial](https://techglimpse.com/install-softhsm-source-compilation/).

**Comment by sberyozkin** [comment]
@gsmet I'm doing a minor follow up update to support configuring such providers but it won't work in the native mode, I'll ask Foivos if something can be done (re https://github.com/oracle/graal/issues/2552)

**Comment by gsmet** [comment]
@sberyozkin I'm not sure if we should merge this or if you want us to wait?

**Comment by sberyozkin** [comment]
@gsmet, Hi, I'd not mind it going into 2.9.0 CR1 if you are OK with it, it won't work in native yet, but there was an earlier query about configuring `SunPKCS11-NSS-FIPS` and I think this is how it will be done, so we can ask SMEs to give it a try in JVM mode

**Review by gsmet — APPROVED** [review]

**Pull request merged**